src/test/evo_deterministicmns_tests.cpp

```diff
@@ -415,6 +450,109 @@ BOOST_FIXTURE_TEST_CASE(dip3_protx, TestChain400Setup)
     BOOST_CHECK(chainTip->nHeight == nHeight);
     BOOST_CHECK(chainTip->GetBlockHash() != pblock->GetHash());
 
+    // ProUpServ: change masternode IP
+    {
+        const uint256& proTx = dmnHashes[InsecureRandRange(dmnHashes.size())];  // pick one at random
+        auto tx = CreateProUpServTx(utxos, proTx, operatorKeys.at(proTx), 1000, CScript(), coinbaseKey);
+
+        CValidationState dummyState;
+        BOOST_CHECK(CheckProUpServTx(tx, chainTip, dummyState));
+        BOOST_CHECK(CheckTransactionSignature(tx));
+        // also verify that payloads are not malleable after they have been signed
+        auto tx2 = MalleateProUpServTx(tx);
+        BOOST_CHECK(!CheckProUpServTx(tx2, chainTip, dummyState));
```

Comment:
Will be good to check for the returned `bad-protx-sig` state error so we can ensure that it's failing where the test expects and not somewhere else.
```
BOOST_CHECK(dummyState.GetRejectReason() == "bad-protx-sig");
```

Reply:
oops, yeah, same as in all other places. Fixed.
Also changed `CheckProUpServTx` calls in the test to more general `CheckSpecialTx`.